### ISSUE TITLE
FamilyNonEvent: missing namespace and wrong assingment (FAM <-> INDI)?

### ISSUE THREAD

**Issue by ungeahnt** [fisharebest/webtrees#4775]
https://github.com/fisharebest/webtrees/blob/8e1afc648389017cb4e48e677289723c2833c768/app/CustomTags/Gedcom7.php#L73

**Comment by fisharebest** [comment]
This looks like a bug.  Should be `FamilyNonEvent`???

Did you intend to close this?

**Comment by ungeahnt** [comment]
> Did you intend to close this?

Yes, because at second glance I've thought that 'FamilyNonEvent' does the same as 'IndividualNonEvent' and that you have decided to create only one of them. 

If that's not how you intended it, then it's probably a bug.

**Comment by fisharebest** [comment]
The sub-tags are different:
* `AGE`
* `HUSB:AGE` / `WIFE:AGE`

**Comment by fisharebest** [comment]
> The sub-tags are different:

Actually, they are the same - and I cannot remember why I did this!

**Comment by ungeahnt** [comment]
> and I cannot remember why I did this!

What? That you have splitted `NonEvent` to `FamilyNonEvent` / `IndividualNonEvent` or that you have created only one of them?

**Comment by Jefferson49** [comment]
> Actually, they are the same

Yes, it is weird, but the GEDCOM 7 standard really allows all kind of NO events in FAM and INDI. 

```
NO (Did not happen) <g7:NO>
An enumerated value from set g7:enumset-EVEN identifying an event type which did
not occur to the superstructure’s subject.
```
```
<g7:enumset-EVEN>
An event-type tag name, but not the generic EVEN tag.
```

I created a test Gedcom file, [fam_no.zip](https://github.com/fisharebest/webtrees/files/13838985/fam_no.zip),  with the following structures:

```
0 @I1@ INDI
1 NO MARB

0 @F1@ FAM
1 NO GRAD
```

I checked the Gedcom file with the validators [GEDCOM Validator](https://chronoplexsoftware.com/gedcomvalidator/) and [Gedcom Service Programs](https://ofb.hesmer.name/gedserpro_e.html). Both of the validators accepted this kind of NO events.

> and I cannot remember why I did this!

Well, intuitively it makes a lot of sense to have a IndividualNonEvent class for INDI and a FamilyNonEvent class for FAM. Personally, I would follow the webtrees logic, but the standards sees it differently.


**Conclusion:**
For me, the consequence of the discussion seems to be that we would need a **NonEvent class** and assign it as follows in [Gedom7.php](https://github.com/fisharebest/webtrees/blob/main/app/CustomTags/Gedcom7.php#L74)

```
            'FAM:NO'                     => new NonEvent(I18N::translate('Event did not occur')),
            'INDI:NO'                    => new NonEvent(I18N::translate('Event did not occur')),
```

**Comment by fisharebest** [comment]
I've raised this issue with the GEDCOM maintainers.

Also, I can't work out why I set both to be the same.  I'll change them back.

**Comment by fisharebest** [comment]
Fixed in e627d11e96 (forgot to put issue number in commit message)
